baremaps-cli/src/main/java/com/baremaps/cli/OpenApi.java

```diff
@@ -0,0 +1,113 @@
+
+package com.baremaps.cli;
+
+import com.baremaps.openapi.services.*;
+import com.baremaps.postgres.jdbc.PostgresUtils;
+import com.baremaps.server.CorsFilter;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.servicetalk.http.api.BlockingStreamingHttpService;
+import io.servicetalk.http.netty.HttpServers;
+import io.servicetalk.http.router.jersey.HttpJerseyRouterBuilder;
+import io.servicetalk.transport.api.ServerContext;
+import org.glassfish.hk2.utilities.binding.AbstractBinder;
+import org.glassfish.jersey.jackson.JacksonFeature;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.jackson2.Jackson2Config;
+import org.jdbi.v3.jackson2.Jackson2Plugin;
+import org.jdbi.v3.postgres.PostgresPlugin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+import javax.sql.DataSource;
+import javax.ws.rs.ext.ContextResolver;
+import javax.ws.rs.ext.Provider;
+import java.util.concurrent.Callable;
+
+@Command(name = "openapi", description = "Serve the openapi API.")
+public class OpenApi implements Callable<Integer> {
+
+  private static final Logger logger = LoggerFactory.getLogger(OpenApi.class);
+
+  @Option(
+      names = {"--database"},
+      paramLabel = "DATABASE",
+      description = "The JDBC url of the Postgres database.",
+      required = true)
+  private String database;
+
+  @Option(
+      names = {"--port"},
+      paramLabel = "PORT",
+      description = "The port of the server.")
+  private int port = 9000;
+
+  @Override
+  public Integer call() throws Exception {
+    DataSource datasource = PostgresUtils.datasource(this.database);
+
+    Jdbi jdbi = Jdbi.create(datasource)
+            .installPlugin(new PostgresPlugin())
+            .installPlugin(new Jackson2Plugin());
+
+    // Configure ObjectMapper to skip None/NULL
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    jdbi.getConfig(Jackson2Config.class).setMapper(mapper);
+
+    // Initialize the application
+    ResourceConfig application = new ResourceConfig()
+            .packages("org.glassfish.jersey.examples.jackson")
+            .registerClasses(MyObjectMapperProvider.class,
```

Comment:
If I understand correctly, this is needed when the webstack (servicetalk) is serializing or deserializing json payloads.

https://apple.github.io/servicetalk//servicetalk-data-jackson-jersey/0.31/index.html

What we probably want is to use the object mapper in service talk and JDBI. Also, we probably want to rename the MyObjectMapperProvider into something more meaningfull. I created an issue (#320) to ensure that we address this point soon, but I don't think it's a show stopper for merging.

Reply:
Yeah i tried this way but could not make it work. Maybe we can have a look at it together this week.